src/univariate/discrete/geometric.jl

```diff
@@ -69,17 +69,8 @@ entropy(d::Geometric) = (-xlogx(succprob(d)) - xlogx(failprob(d))) / d.p
 
 ### Evaluations
 
-function pdf(d::Geometric{T}, x::Int) where T<:Real
-    if x >= 0
-        p = d.p
-        return p < one(p) / 10 ? p * exp(log1p(-p) * x) : d.p * (one(p) - p)^x
-    else
-        return zero(T)
-    end
-end
-
-function logpdf(d::Geometric{T}, x::Int) where T<:Real
-    x >= 0 ? log(d.p) + log1p(-d.p) * x : -T(Inf)
+function logpdf(d::Geometric, x::Real)
```

Comment:
This gives nonsensical results for non-integer x

Reply:
Good catch! Seems I forgot to change the check `x >= 0` to `insupport(d, x)`.

Reply:
I fixed it.